tools/interactive/interactivetool_bam_iobio.xml

```diff
@@ -1,46 +1,43 @@
-<tool id="interactive_tool_bam_iobio" tool_type="interactive" name="BAM (iobio) Visualisation" version="0.1">
+<tool id="interactive_tool_bam_iobio" tool_type="interactive" name="bam.iobio visualisation" version="0.2">
     <requirements>
-        <container type="docker">qiaoy/iobio-bundle.bam-iobio:1.0-ondemand</container>
+        <container type="docker">anderspitman/bam.iobio.io:latest</container>
     </requirements>
     <entry_points>
-        <entry_point name="BAM io.bio visualisation of $infile.display_name" requires_domain="True">
-            <port>80</port>
-            <url><![CDATA[/?bam=http://localhost/tmp/bamfile.bam&region=1]]></url>
+        <entry_point name="BAM io.bio visualisation of $baminfile.display_name" requires_domain="True">
+            <port>9001</port>
+            <url>/</url>
         </entry_point>
     </entry_points>
     <command><![CDATA[
-    ## ToDo: websocket could not be found
-    ## WebSocket connection to 'ws://localhost/bamreaddepther/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
-
-    #set $PUB_HOSTNAME = 'localhost'
-    #set $PUB_HTTP_PORT = '80'
-
-    cd /var/www/html &&
-    sed -i "s@\"wss://services.iobio.io/samtools/\"@((window.location.protocol === \"https:\") ? \"wss://\" : \"ws://\") + window.location.host + \"/samtools/\"@" js/bam.iobio.js/bam.iobio.js &&
-    sed -i "s@\"wss://services.iobio.io/bamreaddepther/\"@((window.location.protocol === \"https:\") ? \"wss://\" : \"ws://\") + window.location.host + \"/bamreaddepther/\"@" js/bam.iobio.js/bam.iobio.js &&
-    sed -i "s@\"wss://services.iobio.io/bamstatsalive/\"@((window.location.protocol === \"https:\") ? \"wss://\" : \"ws://\") + window.location.host + \"/bamstatsalive/\"@" js/bam.iobio.js/bam.iobio.js &&
-    sed -i "s@\"wss://services.iobio.io/samheader/\"@((window.location.protocol === \"https:\") ? \"wss://\" : \"ws://\") + window.location.host + \"/samheader/\"@" js/bam.iobio.js/bam.iobio.js &&
-
-    sed -i 's/deny all;//g' /etc/nginx/nginx.conf &&
-
-        cp '${infile}' /input/bamfile.bam &&
-        cp '${infile.metadata.bam_index}' /input/bamfile.bam.bai &&
```

Comment:
@luke-c-sargent bai files are associated to BAM files. So you can use that trick for your inputs as well.

Reply:
are these bam indices generated when importing a bam file (no bai in history required), or does does metadata entry get set to a bai file in your history? 

i ask because the bai file is a hard requirement so including it as a parameter was meant to make that requirement explicit, but i am happy to remove it if it's not necessary as it makes it more user-friendly to only need the bam file

Reply:
Yes, Galaxy should always generate a .bai (or the newer alternative for specific depths, I think?) for these files.

Reply:
now using the metadata bai (neat!), making it just one input